File: src/main/java/com/github/ericufo/jedai/chat/Answer.java
package com.github.ericufo.jedai.chat;

import com.github.ericufo.jedai.rag.RetrievedChunk;

import java.util.List;

/**
 * 生成的答案
 */
public class Answer {
    private final String content;
    private final List<RetrievedChunk> citations;
    private final boolean isGeneralKnowledge;

    public Answer(String content, List<RetrievedChunk> citations, boolean isGeneralKnowledge) {
        // 验证：如果有引用，则不应标记为通用知识
        if (!citations.isEmpty() && isGeneralKnowledge) {
            throw new IllegalArgumentException("如果有引用，则不应标记为通用知识");
        }
        this.content = content;
        this.citations = citations;
        this.isGeneralKnowledge = isGeneralKnowledge;
    }

    public String getContent() {
        return content;
    }

    public List<RetrievedChunk> getCitations() {
        return citations;
    }

    public boolean isGeneralKnowledge() {
        return isGeneralKnowledge;
    }
}

File: src/main/kotlin/com/github/ericufo/jedai/chat/Answer.kt
package com.github.ericufo.jedai.chat

import com.github.ericufo.jedai.rag.RetrievedChunk

/**
 * 生成的答案
 */
data class Answer(
    val content: String,
    val citations: List<RetrievedChunk>,
    val isGeneralKnowledge: Boolean  // true表示基于通用知识，false表示基于课程材料
) {
    init {
        // 验证：如果有引用，则不应标记为通用知识
        require(!(citations.isNotEmpty() && isGeneralKnowledge)) {
            "如果有引用，则不应标记为通用知识"
        }
    }
}

